src/HTMLTextLine.cc

```diff
@@ -269,7 +280,10 @@ void HTMLTextLine::dump_text(ostream & out)
                             double threshold = state_iter1->em_size() * (param.space_threshold);
 
                             out << "<span class=\"" << CSS::WHITESPACE_CN
-                                << ' ' << CSS::WHITESPACE_CN << wid << "\">" << (target > (threshold - EPS) ? " " : "") << "</span>";
+                                << ' ' << CSS::WHITESPACE_CN << wid << "\">";
+                            if (target > (threshold - EPS))
+                                dump_unicode(out, ' ');
```

Comment:
The space is in that span, not after it. This change doesn't change previous behavior, just record `last_output_unicode`.